client/src/actions/userActions.js

```diff
@@ -0,0 +1,77 @@
+import axios from 'axios';
+import {setAlert} from './alert';
+
+import { USER_DETAILS_FAIL, USER_DETAILS_REQUEST, USER_DETAILS_SUCCESS, USER_LOGIN_FAIL, USER_LOGIN_REQUEST, USER_LOGIN_SUCCESS, USER_REGISTER_FAIL, USER_REGISTER_REQUEST, USER_REGISTER_SUCCESS } from '../constants/userConstants';
+
+export const login = (email,password, history) => async (dispatch) =>{
+  dispatch({type:USER_LOGIN_REQUEST});
+
+    try {
+        const config = {
+            headers:{
+                'Content-Type': 'application/json'
+            }
+        }
+        await axios.post('/users/login',{email, password},config);
+        dispatch({type:USER_LOGIN_SUCCESS})
+        dispatch(loadUser());
+        history.push('/dashboard');
+    } catch (error) {
+      dispatch({type:USER_LOGIN_FAIL})
+      const errors = error.response.data.errors;
+       if (errors){
+           errors.forEach(err=>dispatch(setAlert('login',err.msg)));
+         }
+    }
+}
+
+export const register = (name, email,password, history) => async (dispatch) =>{
+  dispatch({type:USER_REGISTER_REQUEST});
+
+  try {
+      const config = {
+          headers:{
+              'Content-Type': 'application/json'
+          }
+      }
+      await axios.post('/users/register',{name, email, password},config);
+
+      dispatch({type:USER_REGISTER_SUCCESS});
+      dispatch(setAlert('register','register successful!','success'));
```

Comment:
yes sorry, I forgot to delete this line. This was the first implementation I had that I wanted to have success alerts too, which I changed after, I thought we don't need them since the page will change anyway. I modified the file accordingly. This line I forgot. I will delete it.